src/groupederror.jl

```diff
@@ -0,0 +1,178 @@
+get_axis(column) = sort!(union(column))
+get_axis(column, npoints) = linspace(minimum(column),maximum(column),npoints)
+
+# f is the function used to analyze dataset: define it as nan when it is not defined,
+# the input is: dataframe used, points chosen on the x axis, x (and maybe y) column labels
+# the output is the y value for the given xvalues
+
+"""
+    get_mean_sem(trend,variation, f, splitdata::GroupedDataFrame, xvalues, args...; kwargs...)
+
+Apply function `f` to `splitdata`, then output summary statistics
+`trend` and `variation` of those values. A shared x axis `xvalues` is needed: use
+`LinSpace` for continuous x variable and a normal vector for the discrete case. Remaining arguments
+are label of x axis variable and extra arguments for function `f`. `kwargs...` are passed
+to `f`
+"""
+function get_mean_sem(trend,variation, f, splitdata::GroupedDataFrame, xvalues, args...; kwargs...)
+    v = Array(Float64, length(xvalues), length(splitdata));
+    for i in 1:length(splitdata)
+        v[:,i] = f(splitdata[i],xvalues, args...; kwargs...)
+    end
+    mean_across_pop = Array(Float64, length(xvalues));
+    sem_across_pop = Array(Float64, length(xvalues));
+    for j in 1:length(xvalues)
+        notnan = !isnan(v[j,:])
+        mean_across_pop[j] = trend(v[j,notnan])
+        sem_across_pop[j] = variation(v[j,notnan])
+    end
+    valid = !isnan(mean_across_pop) & !isnan(sem_across_pop)
+    return xvalues[valid], mean_across_pop[valid], sem_across_pop[valid]
+end
+
+"""
+    get_mean_sem(trend,variation, f, df::AbstractDataFrame, across, axis_type, args...; kwargs...)
+
+Split `df` by `across`, choose shared axis according to `axis_type`
+(`:continuous` or `:discrete`) then compute `get_mean_sem`
+"""
+function get_mean_sem(trend,variation, f, df::AbstractDataFrame, across, axis_type, args...; kwargs...)
+    # define points on x axis
+    if axis_type == :discrete
+        xvalues = get_axis(df[args[1]])
+    elseif axis_type == :continuous
+        xvalues = get_axis(df[args[1]], 100)
+    else
+        error("Unexpected axis_type: only :discrete and :continuous allowed!")
+    end
+
+    if across == []
+        mean_across_pop = f(df,xvalues, args...; kwargs...)
+        sem_across_pop = zeros(length(xvalues));
+        valid = ~isnan(mean_across_pop)
+        return xvalues[valid], mean_across_pop[valid], sem_across_pop[valid]
+    else
+        # get mean value and sem of function of interest
+        splitdata = groupby(df, across)
+        return get_mean_sem(trend,variation, f, splitdata, xvalues, args...; kwargs...)
+    end
+end
+
+"""
+    groupapply(f::Function, df, args...; axis_type = :auto, across = [], group = [], summarize = (mean, sem), kwargs...)
+
+Split `df` by `group`. Then apply `get_mean_sem` to get a population summary of the grouped data.
+Output is a `GroupedError` with population summaries, then can be plot using `plot(g::GroupedError)`
+Seriestype can be specified to be `:path`, `:scatter` or `:bar`
```

Comment:
There are a few typos here. Also, the `summarize = (mean, sem)` argument seems to be at odds with "Then apply `get_mean_sem`", as that would change with different `summarize`, right?